### PR TITLE
DBZ-6670 Replace error handler instead of recreate in order to keep retry counter

### PR DESCRIPTION
**Pull request by jcechace**
https://issues.redhat.com/browse/DBZ-6670

### CODE REVIEW

**Comment by jcechace** [comment]
Still missing tests as well as non-core connectors.

File: debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/MongoDbConnectorTask.java [hunk 1]
```diff
@@ -97,7 +97,12 @@ public ChangeEventSourceCoordinator<MongoDbPartition, MongoDbOffsetContext> star
                     .loggingContextSupplier(() -> taskContext.configureLoggingContext(CONTEXT_NAME))
                     .build();
 
-            errorHandler = new MongoDbErrorHandler(connectorConfig, queue);
```
Comment (jpechane) [inline debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/MongoDbConnectorTask.java:100]: @jcechace  How about using `errorHandler = new MongoDbErrorHandler(connectorConfig, queue, errorHandler);` ?

We could use the same approach for `SqlServerErrorHandler` too.

**Comment by jcechace** [comment]
DB2: https://github.com/debezium/debezium-connector-db2/pull/114
Vitess: https://github.com/debezium/debezium-connector-vitess/pull/146
Spanner: https://github.com/debezium/debezium-connector-spanner/pull/47

**Comment by jpechane** [comment]
@jcechace LGTM, applied, thanks!

**Comment by jpechane** [comment]
@jcechace Feel free to backport the PR yourself

**Comment by jcechace** [comment]
@jpechane backported. Thanks!

**Pull request merged**